handsontable/src/core.js

```diff
@@ -138,7 +136,23 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
    */
   this.executionSuspendedCounter = 0;
 
-  keyStateStartObserving(this.rootDocument);
+  const shortcutManager = createShortcutManager({
+    beforeKeyDown: (event) => {
+      if (this.isListening() === false) { // Performing action (executing a callback) and triggering hook only for listening Handsontable.
+        return false;
+      }
+
+      return this.runHooks('beforeKeyDown', event);
+    },
+    afterKeyDown: (event) => {
+      if (this.isDestroyed) { // Handsontable could be destroyed after performing action (executing a callback).
+        return;
+      }
+
+      this.runHooks('afterDocumentKeyDown', event);
+    },
+    frame: this.rootWindow,
```

Comment:
WDYT to changing the name to `ownerWindow`?
```suggestion
    ownerWindow: this.rootWindow,
```